### PR TITLE
Experiments Manager (step 1) - decouple lock

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Description ###

Decouple lock stuffs from experiments manager logic, so we can later switch the lock implementation.

### Checklist ###
  - [x] test case
  - ~~doc~~

### CODE REVIEW

File: ts/nni_manager/test/extensions/experimentManager.test.ts [hunk 1]
```diff
@@ -1,60 +1,56 @@
...
+}
 
+describe('Unit test for experiment manager', function () {
     before(() => {
         prepareUnitTest();
-        fs.writeFileSync('.experiment.test', JSON.stringify(mockedInfo));
-        Container.bind(ExperimentManager).to(NNIExperimentsManager).scope(Scope.Singleton);
-        experimentManager = component.get(NNIExperimentsManager);
-        experimentManager.setExperimentPath('.experiment.test');
+        tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'nni-ut-'));
```
Comment (J-shang) [inline ts/nni_manager/test/extensions/experimentManager.test.ts:34]: `nni-ut-`?

Comment (J-shang) [inline ts/nni_manager/test/extensions/experimentManager.test.ts:34]: `nni-ut-`?

**Review by J-shang — APPROVED** [review]
LGTM

File: ts/nni_manager/extensions/experiments_manager/index.ts [hunk 2]
```diff
@@ -1,4 +1,24 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-export { NNIExperimentsManager } from './manager';
+import { ExperimentsManager } from './manager';
+export { ExperimentsManager } from './manager';
+
+let singleton: ExperimentsManager | null = null;
```
Comment (ultmaster) [inline ts/nni_manager/extensions/experiments_manager/index.ts:7]: Is this a real singleton?

File: ts/nni_manager/test/extensions/experimentManager.test.ts (hunk 1 above)
Comment (liuzhe-lz) [reply ts/nni_manager/test/extensions/experimentManager.test.ts:34]: This is directory fix, `tempDir` will be like `/tmp/nni-ut-ri13j`

File: ts/nni_manager/extensions/experiments_manager/index.ts (hunk 2 above)
Comment (liuzhe-lz) [reply ts/nni_manager/extensions/experiments_manager/index.ts:7]: Yes, each NNI manager process will have only one experiments manager.

Comment (liuzhe-lz) [reply ts/nni_manager/extensions/experiments_manager/index.ts:7]: It's not exported, so the only way to access it is `getExperimentManager()`.

**Pull request merged**